### PR TITLE
Fix some problems with modules

### LINKED ISSUE #5818: [2.3] 'Configure' button for adjusting Device Interface fields does nothing
### Environment
* Nautobot version (Docker tag too if applicable): `b49429ebca31 (v2.3.0a1)`
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Navigate to a device which has interfaces on https://next.demo.nautobot.com/ (e.g. [This one](https://next.demo.nautobot.com/dcim/devices/37a938b0-bd5a-4c25-9d98-a51c75d15ba9/interfaces/))
2. Attempt to click this button:
Screenshot 2024-06-14 at 2 52 44 PM

### Expected Behavior
The appearance of the modal "Table Configuration" box, such as this example taken from demo.nautobot.com
Screenshot 2024-06-14 at 3 02 56 PM

### Observed Behavior
Nothing happens. Additionally, using the browser's Web Developer Tools, no request appears to be made when the button is clicked.

Per discussion in Slack:

Glenn M [9 minutes ago](https://networktocode.slack.com/archives/C01NWPK6WHL/p1718391344262159?thread_ts=1718387850.837079&cid=C01NWPK6WHL)
button has data-target="#DeviceInterfaceTable_config" but the table is now id="DeviceModuleInterfaceTable_config"

### LINKED ISSUE #5841: Column role is missing under device -> interfaces (and the configure button does nothing)
### Environment

* Nautobot version: `next`
* Python version: `3.9.6`
* Database platform, version: Postgres
* Middleware(s):

In next version, https://github.com/nautobot/nautobot/pull/5488#issuecomment-2187099350 is implementing interface roles. In the interface table, under a specific device, role is not displayed and the `configure` button does nothing.

### Steps to Reproduce

1. Go to: https://next.demo.nautobot.com/dcim/devices/eaeca318-23f3-4ef0-805c-fceb9d23a0dd/interfaces/ (ams01-leaf-01)
2. Go to Interfaces
3. See `Roles` is missing from the table and pressing `configure` does nothing.

### Expected Behavior

I expect the `configure` button help me to add the `role` column. Also the `role` column should be displayed by default.

### Observed Behavior

`Roles` is missing from the interface table (under a specific device) table and pressing `configure` does nothing.

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes #5818, Closes #5841
# What's Changed

I found some minor issues when preparing for a demo. See screenshots below.

# Screenshots

### Before
image

### After

image

### Before

image

### After

image

## IPAM IPAddress form

This form was broken because it was trying to use `interface.device.location` but the `device` field is optional now so I added the checks for `interface.device is not None`, etc.

### Before

image

### After

image

## Configure table buttons for all device and module component tabs

I broke this button in #5657 when I renamed the table classes. I added templatetags for `table_config_button` and `table_config_button_small` to help prevent this issue going forward.

image

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/5880.fixed [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Fixed overflowing device component tables in device and module component tabs.
+Fixed broken table configure buttons in device and module component tabs.
```
Comment (glennmatthews) [inline changes/5880.fixed:2]: This fixes #5818, I believe?

File: nautobot/dcim/views.py [hunk 2]
```diff
@@ -1931,7 +1931,6 @@ def get_extra_context(self, request, instance):
                 "tags",
             )
             .select_related("lag", "cable")
-            .order_by("_name")
         )
         interface_table = tables.DeviceModuleInterfaceTable(data=interfaces, user=request.user, orderable=False)
         if VirtualChassis.objects.filter(master=instance).exists():
```
Comment (glennmatthews) [inline nautobot/dcim/views.py:1936]: Can we add an `interface_table.columns.show("role")` around here to address #5841?

File: nautobot/dcim/templates/dcim/module_powerports.html [hunk 3]
```diff
@@ -11,11 +10,11 @@
                 <strong>Power Ports</strong>
                 <div class="pull-right noprint">
                     {% if request.user.is_authenticated %}
-                        <button type="button" class="btn btn-default btn-xs" data-toggle="modal" data-target="#ModulePowerPortTable_config" title="Configure table"><i class="mdi mdi-cog"></i> Configure</button>
+                        {% table_config_button_small powerport_table %}
```
Comment (glennmatthews) [inline nautobot/dcim/templates/dcim/module_powerports.html:13]: Yay for template tags meaning less boilerplate!

File: changes/5880.fixed [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Fixed overflowing device component tables in device and module component tabs.
```
Comment (glennmatthews) [inline changes/5880.fixed:1]: Any chance a similar fix is needed for VirtualMachine/VMInterface?

File: nautobot/dcim/views.py [hunk 5]
```diff
@@ -1931,7 +1931,6 @@ def get_extra_context(self, request, instance):
                 "tags",
             )
             .select_related("lag", "cable")
-            .order_by("_name")
```
Comment (glennmatthews) [inline nautobot/dcim/views.py:1934]: What was incorrect about this?

File: changes/5880.fixed (hunk 1 above)
Comment (gsnider2195) [reply changes/5880.fixed:2]: Looks like it. I'll adjust the changelog

File: nautobot/dcim/views.py (hunk 5 above)
Comment (gsnider2195) [reply nautobot/dcim/views.py:1934]: There's something weird going on with our `_name` field. I'll try to see if I can write a test for it and fix it.

```
>>> for i in modular_device.all_interfaces.values("name", "_name"):
...     print(i["name"])
...     print(i["_name"])
... 
mgmt0
9999999999999999mgmt000000............
TenGigabitEthernet1/2/1
0001000299999999TenGigabitEthernet000001............
TenGigabitEthernet1/2/2
0001999999999999TenGigabitEthernet..................{module}/00000002
TenGigabitEthernet1/2/3
0001999999999999TenGigabitEthernet..................{module}/00000003
TenGigabitEthernet1/2/4
0001999999999999TenGigabitEthernet..................{module}/00000004
Ethernet1/4/1
0001000499999999Ethernet000001............
Ethernet1/4/2
0001999999999999Ethernet..................{module}/00000002
Ethernet1/5/1
0001999999999999Ethernet..................{module}/00000001
Ethernet1/5/2
0001999999999999Ethernet..................{module}/00000002
>>> for i in modular_device.all_interfaces.all():
...     i.save()
... 
>>> for i in modular_device.all_interfaces.values("name", "_name"):
...     print(i["name"])
...     print(i["_name"])
... 
mgmt0
9999999999999999mgmt000000............
TenGigabitEthernet1/2/1
0001000299999999TenGigabitEthernet000001............
TenGigabitEthernet1/2/2
0001000299999999TenGigabitEthernet000002............
TenGigabitEthernet1/2/3
0001000299999999TenGigabitEthernet000003............
TenGigabitEthernet1/2/4
0001000299999999TenGigabitEthernet000004............
Ethernet1/4/1
0001000499999999Ethernet000001............
Ethernet1/4/2
0001000499999999Ethernet000002............
Ethernet1/5/1
0001000599999999Ethernet000001............
Ethernet1/5/2
0001000599999999Ethernet000002............
```

Comment (glennmatthews) [reply nautobot/dcim/views.py:1934]: Ew, something where it's not liking the `{module}` syntax huh?

File: changes/5880.fixed (hunk 4 above)
Comment (gsnider2195) [reply changes/5880.fixed:1]: That one is already good. Thanks for the reminder to check it!

File: nautobot/dcim/views.py (hunk 5 above)
Comment (gsnider2195) [reply nautobot/dcim/views.py:1934]: Looks to me like I optimized the model save a little bit too much [here](https://github.com/nautobot/nautobot/blob/cd6bff67514b148293fe37c7fa1594e29d03f115/nautobot/dcim/models/device_components.py#L182)

File: changes/5880.fixed [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+Fixed an exception when trying to edit an IPAddress that had a NAT Inside IPAddress that was related to a component attached to a module.
+Fixed incorrect sort order of interfaces in the device and module interface tabs.
```
Comment (glennmatthews) [inline changes/5880.fixed:3]: remove?

```suggestion
```

**Review by glennmatthews — APPROVED** [review]

File: changes/5880.fixed (hunk 6 above)
Comment (glennmatthews) [reply changes/5880.fixed:3]: Or perhaps not.

**Pull request merged**